File: documentation-examples-java/src/java/dev/typr/foundations/docs/sqlite/NumericTypes.java
package dev.typr.foundations.docs.sqlite;

import dev.typr.foundations.SqliteType;
import dev.typr.foundations.SqliteTypes;
import java.math.BigDecimal;

@SuppressWarnings("unused")
public class NumericTypes {
  // start
  // SQLite stores BigDecimal as TEXT to preserve precision — declared (p,s) is a label only.
  SqliteType<BigDecimal> numericType = SqliteTypes.numeric;
  SqliteType<BigDecimal> decimalType = SqliteTypes.decimal;
  SqliteType<BigDecimal> precise = SqliteTypes.decimalOf(18, 6);
  // stop
}

File: documentation-examples-scala/src/scala/dev/typr/foundationssc/docs/sqlite/NumericTypes.scala
package dev.typr.foundationssc.docs.sqlite
import dev.typr.foundationssc.*

@SuppressWarnings(Array("unused"))
object NumericTypes:
  // start
  val numericType: SqliteType[BigDecimal] = SqliteTypes.numeric
  val decimalType: SqliteType[BigDecimal] = SqliteTypes.decimal
  val precise: SqliteType[BigDecimal] = SqliteTypes.decimalOf(18, 6)
  // stop
